src/robot/libdocpkg/model.py

```diff
@@ -190,3 +228,155 @@ def _arg_to_dict(self, arg):
             'required': arg.required,
             'repr': unicode(arg)
         }
+
+
+class DataTypeCatalog(object):
+
+    def __init__(self):
+        self._enum_catalog = set()
+        self._typed_dict_catalog = set()
```

Comment:
I've started to prefer not prefixing attributes with `_` in cases like this and think `_catalog` postfix is unnecessary as well. I'd thus use just `self.enums` and `self.typed_dicts`, but if you prefer the current variants that's fine as well.